### DISCUSSION TITLE
How does svelte kit get routing configuration

### DISCUSSION THREAD

**Discussion by baiwusanyu-c** [sveltejs/kit#10308] [Q&A] [answered]
I want to get the routing information (such as path, parameters, etc.) of the project through js, but I don't seem to find the relevant methods and APIs in the documentation, what should I do?

**Comment by rzoleta** [comment, marked as answer]
Do you mean from the `<script>` tag of a `.svelte` file?

You can import the `page` store from `$app/stores` then call it to get the params and route id:

```html
<script lang="ts">
    import {page} from `$app/stores`;

</script>

<h1>Route: {$page.route.id}</h1>
<h1>Params: {JSON.stringify($page.url.searchParams.getAll('hello'))}</h1>
```

![CleanShot 2023-07-03 at 22 45 57](https://github.com/sveltejs/kit/assets/18186677/1e1f9a61-a9f6-48d0-a56b-b193c80f502c)

https://learn.svelte.dev/tutorial/page-store
